docs/getting-started/rulesets.md

```diff
@@ -199,3 +198,141 @@ rules:
 The example above will run the recommended rules from the `spectral:oas2` ruleset, but report `operation-2xx-response` as a warning rather than as an error (as is the default behavior in the `spectral:oas2` ruleset).
 
 Available severity levels are `error`, `warn`, `info`, `hint`, and `off`.
+
+
+### Adding a custom function
+
+If the built-in functions are not enough, Spectral gives you a possibility of providing your own ones.
+Previously, this used to be possible only in case of programmatic usage of Spectral, but there is no such limitation anymore,
+and functions are an integral part of Spectral rulesets.
+
+A custom function might be any JS function compliant with [IFunction](https://github.com/stoplightio/spectral/blob/90a0864863fa232bf367a26dace61fd9f93198db/src/types/function.ts#L3#L8) type.
+
+```ts
+export type IFunction<O = any> = (
+  targetValue: any,
+  options: O,
+  paths: IFunctionPaths,
+  otherValues: IFunctionValues,
+) => void | IFunctionResult[];
+```
+
+#### Introduction
```

Comment:
We can't have an introduction that starts on line 220. Maybe this will be improved by moving custom functions to their own document, but an introduction should just be the start of writing about a topic and doesnt need a header.